### ISSUE TITLE
Send data to APICURON (placeholder issue)

### ISSUE THREAD

**Issue by kimrutherford** [pombase/pombase-chado#1013]
I don't think we have an issue specifically for this: we need to send curation data to APICURON.

While implementing pombase/website#525, I wrote the code to collect the basic curation stats for the annotation from Canto.  So that's a start.

We now need to:
  - collect those stats for the other annotation especially HTP  annotation that can be associated with a community curator
  - work out how to send the data using the APICURON API

We need to add more detail to that list.  :-)

**Comment by ValWood** [comment]
Yes we will speak with them again at some point soon -ish. MAybe OCt/Nov ish. Let's get other stuff out of the way first.

**Comment by kimrutherford** [comment]
I've just emailed them to ask how to make "Terms" on apicuron.org as described in the "Defining your Terms" section of the [docs](https://dev.apicuron.org/docs).  I can't find the dialog on the [dev site](dev.apicuron.org).

**Comment by kimrutherford** [comment]
> I've just emailed them to ask how to make "Terms" 

Turns out there was a bug that they've helped me work around.

**Comment by kimrutherford** [comment]
I've changed the nightly Chado processing code to make a file for testing the APICURON API.  After a few false starts I've been able to submit the file to the dev APICURON server.

The test file contains a record for each time a user has managed to get a publication/session approved.  Each record looks like:
```json
    {
      "activity_term": "publication_curated",
      "timestamp": "2022-02-24T11:54:36.000Z",
      "curator_orcid": "0000-0003-4148-4606",
      "entity_uri": "https://www.pombase.org/reference/PMID:15964794"
    }
```

Let's talk about this on Zoom/Skype.  I'd like to see what happens when you try to log in to the dev server.

--------------------------------------------------------

Note to self, example of how to post Curl using:

```sh
curl -X 'POST' \
  'https://dev.apicuron.org/api/reports/bulk' \
  -H 'accept: */*' \
  -H 'version: 2' \
  -H 'authorization: bearer ...' \
  -H 'Content-Type: multipart/form-data' \
  -F 'delete_all=pombase' \                                   
  -F 'reports=@apicuron_data.json;type=application/json' 
```

**Comment by kimrutherford** [comment]
Current status (2023-10-09): I've reported a few bugs via the APICURON Slack (on 2023-10-5) and I'm waiting for replies.

**Comment by kimrutherford** [comment]
> Current status (2023-10-09): I've reported a few bugs via the APICURON Slack (on 2023-10-5) and I'm waiting for replies.

Those bugs are fixed.

**Comment by kimrutherford** [comment]
As an experiment, I've added a command at the very end of the nightly load script to attempt to send our latest data to APICURON.  I've put it at the end of the script so that if there are any problems it won't cause other load steps to be missed.

I'll check on Wednesday to see if that works OK.

Note that the data is still very minimal: just records of which ORCID curated which paper.

Val, when you have time (maybe next week?) we should try to get you logged in on the dev APICURON server to see how it looks to you.

**Comment by ValWood** [comment]
Yes, let's do this next week.

**Comment by kimrutherford** [comment]
> I've put it at the end of the script so that if there are any problems it won't cause other load steps to be missed.

The load script fell over just before trying to upload to APICURON because I made a typo implementing pombase/pombase-chado#886.  It's fixed now and I'll check the APICURON upload on Thursday morning.

**Comment by kimrutherford** [comment]
> I've put it at the end of the script so that if there are any problems it won't cause other load steps to be missed.

That's working now so the PomBase data in the dev APICURON is now updated nightly.

**Comment by kimrutherford** [comment]
APICURON gives has a graph of contributions:

![image](https://github.com/pombase/pombase-chado/assets/90474/c181ac8b-70f2-418f-9307-b98d7dca0a18)

**Comment by kimrutherford** [comment]
I've added a second APICURON "Activity Term" for when a user submits a HTP dataset.  I want to see how it looks in the interface if a user has two different types of record.  I've changed the code the generates the JSON too.

The new term ID is "provided_dataset".  The existing term ID is "publication_curated".  These are just terms for testing on the dev server.  When we move other to the main apicuron.org we can rename or replace them.

**Comment by kimrutherford** [comment]
From the call last week:
  - export the a new type `approved_session` so that admins get some recognition

**Comment by kimrutherford** [comment]
> export the a new type approved_session so that admins get some recognition

The approver records are now sent to the dev APICURON too.  We now have three "activity terms":
  - publication_curated (session curated in Canto)
  - provided_dataset (PHAF file)
  - approved_publication (approved a session in Canto)

I think that's quite a good set for this project.
